graph/src/data_source/mod.rs

```diff
@@ -158,9 +158,19 @@ impl<C: Blockchain> DataSource<C> {
         logger: &Logger,
     ) -> Result<Option<TriggerWithHandler<MappingTrigger<C>>>, Error> {
         match (self, trigger) {
-            (Self::Onchain(ds), TriggerData::Onchain(trigger)) => ds
-                .match_and_decode(trigger, block, logger)
-                .map(|t| t.map(|t| t.map(MappingTrigger::Onchain))),
+            (Self::Onchain(ds), TriggerData::Onchain(trigger)) => {
+                // If the data source has an `endBlock` field, check that the trigger block is
+                // within the range of blocks that the data source is supposed to handle.
+                // Otherwise, ignore the trigger.
+                if let Some(end_block) = ds.end_block() {
```

Comment:
Maybe this can be added as another pattern with a guard, feels like that would be a lot easier read. something like: 

```
(Self::Onchain(ds), _ ) if ds.has_expired(block.number())
```

Reply:
This is done